core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -1004,6 +1005,13 @@ public boolean supportsGroupByWithCube() {
     return false;
   }
 
+  /**
+   * Returns whether this dialect supports certain type of join in the underlying DBMS.
+   */
+  public boolean isSupportedCertainJoinType(JoinRelType certainJoinType) {
```

Comment:
Rename to `supportsJoinType` to be uniform with the other methods in this class and the param to be simply `joinType`.